pandas/core/generic.py

```diff
@@ -3751,6 +3752,80 @@ def last(self, offset):
         start = self.index.searchsorted(start_date, side='right')
         return self.ix[start:]
 
+    def rank(self, axis=0, numeric_only=None, method='average',
+             na_option='keep', ascending=True, pct=False):
+        """
+        Compute numerical data ranks (1 through n) along axis. Equal values are
+        assigned a rank that is the average of the ranks of those values
+
+        Parameters
+        ----------
+        axis : int or string, default 0
+        numeric_only : boolean, default None
+            Include only float, int, boolean data. Valid only for DataFrame or
+            Panel objects
+        method : {'average', 'min', 'max', 'first', 'dense'}
+            * average: average rank of group
+            * min: lowest rank in group
+            * max: highest rank in group
+            * first: ranks assigned in order they appear in the array
+            * dense: like 'min', but rank always increases by 1 between groups
+        na_option : {'keep', 'top', 'bottom'}
+            * keep: leave NA values where they are
+            * top: smallest rank if ascending
+            * bottom: smallest rank if descending
+        ascending : boolean, default True
+            False for ranks by high (1) to low (N)
+        pct : boolean, default False
+            Computes percentage rank of data
+
+        Returns
+        -------
+        ranks : same type as caller
+        """
+        axis = self._get_axis_number(axis)
+
+        # this method does not make sense when ndim > 2
+        if self.ndim > 2:
+            raise NotImplementedError
+
+        # numeric_only does not make sense for a Series
+        from pandas import Series
+        if numeric_only and isinstance(self, Series):
+            msg = "numeric_only parameter not valid for Series"
+            raise NotImplementedError(msg)
+
+        # if numeric_only is None, and we can't get anything, we try with
```

Comment:
actually you can easily handle ``numeric_only`` for a ``Series``, IOW. if it numeric (float/int), otherwise just return the Series itself.